src/yad2.py

```diff
@@ -1,49 +1,105 @@
 
 class Yad2:
-	USERNAME_TEXTBOX_ID = 'userName'
-	PASSWORD_TEXTBOX_ID = 'password'
-	SUBMIT_FORM_ID = 'submitLogonForm'
-	ADS_PAGE_LINK_TEXT = u'מכירות'
-	LOGIN_URL = 'https://my.yad2.co.il/login.php'
-
-	def __init__(self):
-		self._driver = webdriver.Chrome(executable_path=CHROME_EXECUTABLE_PATH)
-		self._driver.maximize_window()
-
-	def login(self, email, password):
-		self._driver.get(Yad2.LOGIN_URL)
-		username_textbox = self._driver.find_element_by_id(Yad2.USERNAME_TEXTBOX_ID)
-		password_textbox = self._driver.find_element_by_id(Yad2.PASSWORD_TEXTBOX_ID)
-		submit_button = self._driver.find_element_by_id(Yad2.SUBMIT_FORM_ID)
-		username_textbox.send_keys(email)
-		password_textbox.send_keys(password)
-		submit_button.click()
-
-	def _go_to_ads_page(self):
-		self._driver.find_element_by_partial_link_text(Yad2.ADS_PAGE_LINK_TEXT).click()
-
-	def _open_all_ads(self):
-		rows = self._driver.find_elements_by_tag_name('tr')
-		item_rows = [row for row in rows if len(row.find_elements_by_tag_name('td')) > 3]
-		for item_row in item_rows:
-			item_row.click()
-
-	def bounce_all_ads(self):
-		self._go_to_ads_page()
-		self._open_all_ads()
-
-		frames = self._driver.find_elements_by_tag_name('iframe')
-		for frame in frames:
-
-			self._driver.switch_to.frame(frame)
-			try:
-				btn = self._driver.find_element_by_id('bounceRatingOrderBtn')
-				btn.click()
-				print('Bounced Ad!')
-			except Exception:
-				pass
-
-			self._driver.switch_to.default_content()
+    USERNAME_TEXTBOX_ID = 'userName'
+    PASSWORD_TEXTBOX_ID = 'password'
+    SUBMIT_FORM_ID = 'submitLogonForm'
+    ADS_PAGE_LINK_TEXT = u'מכירות'
+    LOGIN_URL = 'https://my.yad2.co.il/login.php'
+
+    def __init__(self, executable_path):
+        options = webdriver.ChromeOptions()
+        if platform == "linux" or platform == "linux2":
+            options.binary_location = '/usr/bin/google-chrome-stable'
+            options.add_argument('headless')
+        self._driver = webdriver.Chrome(executable_path=executable_path, options=options)
+        logger_handler = logging.FileHandler('yad2.log')
+        logger_formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+        logger_handler.setFormatter(logger_formatter)
+        self._logger = logging.getLogger(__name__)
+        self._logger.setLevel(logging.DEBUG)
+        self._logger.addHandler(logger_handler)
+
+    def login(self, email, password):
+        self._logger.info('Started login')
+        self._driver.get(Yad2.LOGIN_URL)
+        username_textbox = self._driver.find_element_by_id(Yad2.USERNAME_TEXTBOX_ID)
+        password_textbox = self._driver.find_element_by_id(Yad2.PASSWORD_TEXTBOX_ID)
+        submit_button = self._driver.find_element_by_id(Yad2.SUBMIT_FORM_ID)
+        username_textbox.send_keys(email)
+        password_textbox.send_keys(password)
+        submit_button.click()
+        self._logger.info('Finished login')
+
+    def iterate_categories(self):
```

Comment:
Add some documentation for what this generator does.
I think that it isn't clear that a generator also clicks on buttons and changes the page.